### PR TITLE
Feat: Added a default time frame

### LINKED ISSUE #165: Adding default time frame to filtering meter reads
**Objective:**
- To Add a default time frame for filtering meter reads  on the all reads section.

**Description:**
- Add a default time frame for filtering meter reads on the all reads section.

**Acceptance Criteria:**
- Added a default time frame.
- The time frame can be customized.
- Run tests and check if everything works well.
  - The UI project runs without any issue (doesn't cause any regression error)

### PR DESCRIPTION
**Pull request by Kingcedru**
**Objective:**
- To Add a default time frame for filtering meter reads  on the all reads section.

**Description:**
- Add a default time frame for filtering meter reads on the all reads section.

**Acceptance Criteria:**
- Added a default time frame.
- The time frame can be customized.
- Run tests and check if everything works well.
  - The UI project runs without any issue (doesn't cause any regression error)

### CODE REVIEW

**Review by divinecharlotte — DISMISSED** [review]
Looks good for now

File: src/app/view/meter-read/all-metereads/all-metereads.component.ts [hunk 1]
```diff
@@ -31,6 +31,7 @@ export class AllMetereadsComponent implements OnInit {
   dataSource: MatTableDataSource<any>;
   readdata: any;
 
+  startDate: Date = new Date(new Date().setMonth(new Date().getMonth() - 1));
```
Comment (musayann) [inline src/app/view/meter-read/all-metereads/all-metereads.component.ts]: let's not add a new class property for this. We only need this value when initializing

File: src/app/view/meter-read/all-metereads/all-metereads.component.ts [hunk 2]
```diff
@@ -284,7 +285,7 @@ export class AllMetereadsComponent implements OnInit {
       this.externalId = result.externalId;
     }
 
-    this.FilterForm.controls['start'].setValue(result.commissioningDate);
+    this.FilterForm.controls['start'].setValue(this.startDate);
```
Comment (musayann) [inline src/app/view/meter-read/all-metereads/all-metereads.component.ts]: I would suggest instead to set it to 
```ts
this.FilterForm.controls['start'].setValue(result.commissioningDate || startDate);
```

Comment (musayann) [reply src/app/view/meter-read/all-metereads/all-metereads.component.ts]: We should set this date to the startDate that you have calculated as a fallback. The default behavior should be to use the `commissioningDate`

**Review by musayann — CHANGES_REQUESTED** [review]

**Review by musayann — APPROVED** [review]

**Pull request merged**